src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorExtensions.cs

```diff
@@ -0,0 +1,1754 @@
+            }
+
+            Tensor<T> tensor;
+            if (axis == -1)
+            {
+                tensor = new Tensor<T>(values, [valuesCopied], tensors[0].IsPinned);
+            }
+            else
+            {
+                nint[] lengths = new nint[tensors[0].Rank];
+                tensors[0].Lengths.CopyTo(lengths);
+                lengths[axis] = sumOfAxis;
+                tensor = new Tensor<T>(values, lengths, tensors[0].IsPinned);
+            }
+
+            return tensor;
+        }
+
+        private static nint CalculateCopyLength(ReadOnlySpan<nint> lengths, int startingAxis)
+        {
+            // When starting axis is -1 we want all the data at once same as if starting axis is 0
+            if (startingAxis == -1)
+                startingAxis = 0;
+            nint length = 1;
+            for (int i = startingAxis; i < lengths.Length; i++)
+            {
+                length *= lengths[i];
+            }
+            return length;
+        }
+        #endregion
+
+        #region StdDev
+        // REVIEW: ADD IN ONES THAT TAKE AXIS.
+        public static T StdDev<T>(Tensor<T> input)
+            where T : IEquatable<T>, IEqualityOperators<T, T, bool>, IFloatingPoint<T>, IPowerFunctions<T>, IAdditionOperators<T, T, T>, IAdditiveIdentity<T, T>
+
+        {
+            T mean = Mean(input);
+            var span = MemoryMarshal.CreateSpan(ref input._values[0], (int)input._linearLength);
+            var output = new T[input._linearLength].AsSpan();
+            TensorPrimitives.Subtract(span, mean, output);
+            TensorPrimitives.Abs(output, output);
+            TensorPrimitives.Pow((ReadOnlySpan<T>)output, T.CreateChecked(2), output);
+            T sum = TensorPrimitives.Sum((ReadOnlySpan<T>)output);
+            return T.CreateChecked(sum / T.CreateChecked(input.LinearLength));
+        }
+
+        public static Tensor<T> StdDev<T>(Tensor<T> input, int axis)
+            where T : IEquatable<T>, IEqualityOperators<T, T, bool>, IFloatingPoint<T>, IPowerFunctions<T>, IAdditionOperators<T, T, T>, IAdditiveIdentity<T, T>
+            => throw new NotImplementedException();
+
+        public static TResult StdDev<T, TResult>(Tensor<T> input)
+            where T : IEquatable<T>, IEqualityOperators<T, T, bool>, INumber<T>
+            where TResult : IEquatable<TResult>, IEqualityOperators<TResult, TResult, bool>, IFloatingPoint<TResult>
+
+        {
+            T sum = Tensor.Sum(input);
+            return TResult.CreateChecked(TResult.CreateChecked(sum) / TResult.CreateChecked(input.LinearLength));
+        }
+
+        public static Tensor<TResult> StdDev<T, TResult>(Tensor<T> input, int axis)
+            where T : IEquatable<T>, IEqualityOperators<T, T, bool>, INumber<T>
+            where TResult : IEquatable<TResult>, IEqualityOperators<TResult, TResult, bool>, IFloatingPoint<TResult>
+            => throw new NotImplementedException();
```

Comment:
We shouldn't be merging any NotImplemented public APIs. If they're not implemented yet, let's leave them out.

Reply:
KK. Removed them for now.